src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/BlockPlacer.java

```diff
@@ -145,7 +145,10 @@ private boolean isBlacklisted(Material type) {
 
     private void placeSlimefunBlock(SlimefunItem sfItem, ItemStack item, Block block, Dispenser dispenser) {
         BlockPlacerPlaceEvent e = new BlockPlacerPlaceEvent(dispenser.getBlock(), item, block);
-        Bukkit.getPluginManager().callEvent(e);
+        
+        if(dispenser.getInventory().getViewers().isEmpty()) {
+        	Bukkit.getPluginManager().callEvent(e);
```

Comment:
Putting this here makes absolutely no sense.
The only thing it would do is break all other plugins and allow for many other duplications.

Reply:
You cannot just make only the event calling conditional. The check must occur far before any Event is even called, so better before these methods are even invoked in the first place.

Also, you should really test your changes before-hand.